dlclose is a no-op on musl.  Test will always fail.

--- apr-1.6.2/test/testdso.c.old	2010-01-03 19:35:07.000000000 -0600
+++ apr-1.6.2/test/testdso.c	2017-09-10 18:43:43.374983090 -0500
@@ -244,7 +244,7 @@
     abts_run_test(suite, test_load_module, NULL);
     abts_run_test(suite, test_dso_sym, NULL);
     abts_run_test(suite, test_dso_sym_return_value, NULL);
-    abts_run_test(suite, test_unload_module, NULL);
+    /* abts_run_test(suite, test_unload_module, NULL); */
 
 #ifdef LIB_NAME
     apr_filepath_merge(&libname, NULL, LIB_NAME, 0, p);
@@ -252,7 +252,7 @@
     abts_run_test(suite, test_load_library, NULL);
     abts_run_test(suite, test_dso_sym_library, NULL);
     abts_run_test(suite, test_dso_sym_return_value_library, NULL);
-    abts_run_test(suite, test_unload_library, NULL);
+    /* abts_run_test(suite, test_unload_library, NULL); */
 #endif
 
     abts_run_test(suite, test_load_notthere, NULL);
